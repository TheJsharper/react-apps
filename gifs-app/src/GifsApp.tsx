
import GifList from './gifs/components/GifList';
import PreviousSearches from './gifs/components/previousSearches';
import { useGif } from './gifs/hooks/useGif';
import GifsHeader from './shared/components/GifsHeader';
import SearchBar from './shared/components/SearchBar';
const GifsApp = () => {
 const  { gifs, previousSearches, handleSearchClick, handleSearchSubmit } = useGif();
    return (
        <>
            {/* Header */}
            <GifsHeader title='Search Gifs' description='Find the best GIFs for your needs!' />
            {/* Search Form */}
            <SearchBar onSearchSubmit={handleSearchSubmit} placeholder="Search a gif here..." />
            {/* List of Gifs */}
            <PreviousSearches title="Previous Searches" searches={previousSearches}
                onSearchClick={handleSearchClick} />
            <GifList gifs={gifs} />
        </>)
}



export default GifsApp
