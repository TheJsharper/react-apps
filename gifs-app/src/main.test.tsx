import { render, screen } from "@testing-library/react";
import { createRoot } from "react-dom/client";
import { describe, expect, test } from "vitest";
import GifsApp from "./GifsApp";
//import * as mainApp from "./main";
// import "@testing-library/jest-dom" is only needed once in your test setup file, not here
describe('main', () => {

  test('should render correctly', () => {
     const { container } = render(<main />);
        expect(container).toBeDefined();
        screen.debug();
        expect(container).toMatchSnapshot();
  });
  test("should have a root element", () => {
    const component = <GifsApp />;
    const container = document.createElement('div');
    document.body.appendChild(container);

    // ReactDOM.render(component, container);

    const root = createRoot(container);
    root.render(component);
    expect(container.firstChild).toBeDefined();
    screen.debug();

  });

});