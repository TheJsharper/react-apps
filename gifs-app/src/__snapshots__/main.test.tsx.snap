// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`main > should render correctly 1`] = `
<div>
  <main />
</div>
`;

exports[`main > should work 1`] = `
<div>
  <main />
</div>
`;
